Split Coverity scan into a separate test

sgallagher opened the pull request:
Allows us to more easily exclude it for pull-requests and to speed
up the Fedora tests slightly.

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

mmathesius commented on .travis/travis-coverity.sh:
```diff
@@ -0,0 +1,44 @@
+
+rm -f $TARBALL_PATH $SCRIPT_DIR/fedora/Dockerfile.deps.$release $SCRIPT_DIR/fedora/Dockerfile-$release
+
+# Override the standard tasks with the doc-generation
```
doc-generation? (A leftover from `travis-docs.sh`?)

mmathesius commented on .travis/travis-coverity.sh:
```diff
@@ -0,0 +1,44 @@
+    -e TRAVIS=$TRAVIS \
+    -e TRAVIS_JOB_NAME="$TRAVIS_JOB_NAME" \
+    -e TRAVIS_COMMIT="$TRAVIS_COMMIT" \
+    -e DOC_TOKEN="$DOC_TOKEN" \
```
Is `DOC_TOKEN` needed here?

sgallagher replied:
Nope, copy-paste error.

mmathesius commented on .travis/travis-coverity.sh:
```diff
@@ -0,0 +1,44 @@
+
+# Override the standard tasks with the doc-generation
+docker run \
+    -e COVERITY_SCAN_TOKEN=$COVERITY_SCAN_TOKEN \
```
I understand `COVERITY_SCAN_TOKEN` being needed here, but is it (still) needed in `travis-archlinux.sh`, `travis-centos.sh`, `travis-docs.sh`, `travis-fedora.sh`, `travis-mageia.sh`, and `travis-opensuse.sh`?

sgallagher replied:
There's a lot of unnecessary copy-paste going on there, yeah. I'll fix that up.

sgallagher commented:
@mmathesius I updated the patch and also added a follow-up patch to reflow the `docker build` and `docker run` commands in the CI to make them more readable for future maintainers.
